src/cmd/package.go

```diff
@@ -143,6 +144,40 @@ func newPackageCreateCommand(v *viper.Viper) *cobra.Command {
 	return cmd
 }
 
+// Converts registry overrides to a structured type.
+// The result will be sorted in descending order.
+// Descending order guarantees the longest prefix will be sorted toward the beginning.
+//
+// Input is of the following form:
+// []string{"docker.io/library=docker.example.com", "docker.io=docker.example.com"}
+func parseRegistryOverrides(overrides []string) ([]types.RegistryOverride, error) {
+	result := make([]types.RegistryOverride, len(overrides))
+	for i, mapping := range overrides {
+		source, override, found := strings.Cut(mapping, "=")
+		if !found {
+			return nil, fmt.Errorf("invalid registry override: missing '=': %s", mapping)
+		} else if source == "" {
+			return nil, fmt.Errorf("registry override must have a source: %s", mapping)
+		} else if override == "" {
+			return nil, fmt.Errorf("registry override must have a value: %s", mapping)
+		} else if index := slices.IndexFunc(result, func(existing types.RegistryOverride) bool {
+			return existing.Source == source
+		}); index >= 0 {
+			return nil, fmt.Errorf("registry source is duplicated: existing index: %d erroring index: %d source: %s", index, i, source)
+		}
```

Comment:
nit: Just to clean this up and align to updates we're making to the project - flattening and normalizing the error messages would improve the readability. 
```suggestion
        if !found {
            return nil, fmt.Errorf("registry override missing '=': %s", mapping)
        }
        
        if source == "" {
            return nil, fmt.Errorf("registry override missing source: %s", mapping)
        }
        
        if override == "" {
            return nil, fmt.Errorf("registry override missing value: %s", mapping)
        }
        
        if index := slices.IndexFunc(result, func(existing types.RegistryOverride) bool {
            return existing.Source == source
        }); index >= 0 {
            return nil, fmt.Errorf("registry override has duplicate source: existing index %d, new index %d, source %s", index, i, source)
        }
```